lib/runner.js

```diff
@@ -529,7 +533,13 @@ Runner.prototype.runTests = function (suite, fn) {
     }
 
     if (test.isPending()) {
-      self.emit('pending', test);
+      if (self.forbidPending) {
+        test.isPending = alwaysFalse;
```

Comment:
Hmm. The Pending exception is being handled in the `Runner` around [line 533](https://github.com/mochajs/mocha/pull/2874/files#diff-b3b53682a18f203ac8d29b0e277cad26L553), in which a `pending` event is emitted instead of going to the `fail` function at all. (The `fail` function ignores pending tests altogether, rather than emitting anything pending for them; hence the need to circumvent it in this line.)

The `Runnable` is only throwing (and in one case passing to `done`) `Pending` exceptions, not catching any (it also only emits an "error" event, in the case that `done` is called multiple times, so it's not handling the pending emission before sending an ignored error of some sort to the `Runner` either).